consent/handler_test.go

```diff
@@ -300,3 +302,273 @@ func TestGetLoginRequestWithDuplicateAccept(t *testing.T) {
+	}{
+		{
+			desc: "random user_code, not persisted in the database",
+			getBody: func() ([]byte, error) {
+				deviceRequest := fosite.NewDeviceRequest()
+				deviceRequest.Client = cl
+				userCode, _, err := reg.RFC8628HMACStrategy().GenerateUserCode(ctx)
+				require.NoError(t, err)
+				return json.Marshal(&hydra.AcceptDeviceUserCodeRequest{UserCode: &userCode})
+			},
+			getURL: func() string {
+				return ts.URL + "/admin" + DevicePath + "/accept?challenge=" + challenge
+			},
+			validateResponse: func(resp *http.Response) {
+				require.EqualValues(t, http.StatusNotFound, resp.StatusCode)
+			},
+		},
+		{
+			desc: "empty user_code",
+			getBody: func() ([]byte, error) {
+				deviceRequest := fosite.NewDeviceRequest()
+				deviceRequest.Client = cl
+				userCode := ""
+				return json.Marshal(&hydra.AcceptDeviceUserCodeRequest{UserCode: &userCode})
+			},
+			getURL: func() string {
+				return ts.URL + "/admin" + DevicePath + "/accept?challenge=" + challenge
+			},
+			validateResponse: func(resp *http.Response) {
+				require.EqualValues(t, http.StatusBadRequest, resp.StatusCode)
+			},
+		},
+		{
+			desc: "empty challenge",
+			getBody: func() ([]byte, error) {
+				deviceRequest := fosite.NewDeviceRequest()
+				deviceRequest.Client = cl
+				userCode, _, err := reg.RFC8628HMACStrategy().GenerateUserCode(ctx)
+				require.NoError(t, err)
+				return json.Marshal(&hydra.AcceptDeviceUserCodeRequest{UserCode: &userCode})
+			},
+			getURL: func() string {
+				return ts.URL + "/admin" + DevicePath + "/accept"
+			},
+			validateResponse: func(resp *http.Response) {
+				require.EqualValues(t, http.StatusBadRequest, resp.StatusCode)
+			},
+		},
+		{
+			desc: "random challenge",
+			getBody: func() ([]byte, error) {
+				deviceRequest := fosite.NewDeviceRequest()
+				deviceRequest.Client = cl
+				userCode, _, err := reg.RFC8628HMACStrategy().GenerateUserCode(ctx)
+				require.NoError(t, err)
+				return json.Marshal(&hydra.AcceptDeviceUserCodeRequest{UserCode: &userCode})
+			},
+			getURL: func() string {
+				return ts.URL + "/admin" + DevicePath + "/accept?challenge=abc"
+			},
+			validateResponse: func(resp *http.Response) {
+				require.EqualValues(t, http.StatusNotFound, resp.StatusCode)
+			},
+		},
+		{
+			desc: "extra fields",
+			getBody: func() ([]byte, error) {
+				deviceRequest := fosite.NewDeviceRequest()
+				deviceRequest.Client = cl
+				userCode, _, err := reg.RFC8628HMACStrategy().GenerateUserCode(ctx)
+				require.NoError(t, err)
+				ret := struct {
+					UserCode *string
+					Extra    string
+				}{
+					UserCode: &userCode,
+					Extra:    "extra",
+				}
+				return json.Marshal(ret)
+			},
+			getURL: func() string {
+				return ts.URL + "/admin" + DevicePath + "/accept?challenge=" + challenge
+			},
+			validateResponse: func(resp *http.Response) {
+				require.EqualValues(t, http.StatusBadRequest, resp.StatusCode)
+			},
+		},
+	} {
+		t.Run("case="+tc.desc, func(t *testing.T) {
```

Comment:
the issue I reference is [this one](https://go.dev/blog/loopvar-preview)
basically, the closure would grab a hold of the tc, but by the time the closure is run, the tc value is the last iteration value (since the for loop ends, and then the closure start, causing this issue where the last test case is run N times)
You can check this by logging tc values locally to make sure your test is not affected by this.
It's automatically taken care of in Go 1.22 (with some automatic re-scoping of the variable), but we are on 1.21.